topdown/http_test.go

```diff
@@ -2233,6 +2233,148 @@ func TestHTTPSNoClientCerts(t *testing.T) {
+	}
+
+	systemCertsPool, err := x509.SystemCertPool()
+	if err != nil {
+		t.Fatalf("Unexpected error reading system certs %v", err)
+	}
+
+	tempSystemCertsPool, err := x509.SystemCertPool()
+	if err != nil {
+		t.Fatalf("Unexpected error reading system certs %v", err)
+	}
+	systemCertsAndCaPool, err := addCACertsFromBytes(tempSystemCertsPool, ca)
+	if err != nil {
+		t.Fatalf("Unexpected error merging system certs and ca certs %v", err)
+	}
+
+	tests := []struct {
+		note     string
+		input    map[*ast.Term]*ast.Term
+		expected [][]byte
+		msg      string
+		wantNil  bool
+	}{
+		{
+			note:     "tls_use_system_certs set to true",
+			input:    map[*ast.Term]*ast.Term{ast.StringTerm("tls_use_system_certs"): ast.BooleanTerm(true)},
+			expected: systemCertsPool.Subjects(),
+			msg:      "Expected TLS config to use system certs",
+			wantNil:  false,
+		},
+		{
+			note:     "tls_use_system_certs set to nil",
+			input:    map[*ast.Term]*ast.Term{ast.StringTerm("tls_use_system_certs"): ast.BooleanTerm(false)},
+			expected: nil,
+			msg:      "Expected no TLS config",
+			wantNil:  true,
+		},
+		{
+			note:     "no CAs specified",
+			input:    nil,
+			expected: systemCertsPool.Subjects(),
+			msg:      "Expected TLS config to use system certs",
+			wantNil:  false,
+		},
+		{
+			note:     "CA cert provided directly",
+			input:    map[*ast.Term]*ast.Term{ast.StringTerm("tls_ca_cert"): ast.StringTerm(string(ca))},
+			expected: caPool.Subjects(),
+			msg:      "Expected TLS config to use provided CA certs",
+			wantNil:  false,
+		},
+		{
+			note:     "CA cert file path provided",
+			input:    map[*ast.Term]*ast.Term{ast.StringTerm("tls_ca_cert_file"): ast.StringTerm(localCaFile)},
+			expected: caPool.Subjects(),
+			msg:      "Expected TLS config to use provided CA certs in file",
+			wantNil:  false,
+		},
+		{
+			note:     "CA cert provided in env variable",
+			input:    map[*ast.Term]*ast.Term{ast.StringTerm("tls_ca_cert_env_variable"): ast.StringTerm("CLIENT_CA_ENV")},
+			expected: caPool.Subjects(),
+			msg:      "Expected TLS config to use provided CA certs in env variable",
+			wantNil:  false,
+		},
+		{
+			note: "CA cert provided directly and tls_use_system_certs parameter set to false",
+			input: map[*ast.Term]*ast.Term{
+				ast.StringTerm("tls_ca_cert"):          ast.StringTerm(string(ca)),
+				ast.StringTerm("tls_use_system_certs"): ast.BooleanTerm(false),
+			},
+			expected: caPool.Subjects(),
+			msg:      "Expected TLS config to use provided CA certs only",
+			wantNil:  false,
+		},
+		{
+			note: "CA cert provided directly and tls_use_system_certs parameter set to true",
+			input: map[*ast.Term]*ast.Term{
+				ast.StringTerm("tls_ca_cert"):          ast.StringTerm(string(ca)),
+				ast.StringTerm("tls_use_system_certs"): ast.BooleanTerm(true),
+			},
+			expected: systemCertsAndCaPool.Subjects(),
+			msg:      "Expected TLS config to use provided CA certs and system certs",
+			wantNil:  false,
+		},
```

Comment:
I suspect we could combine `note` and `msg`, but what we do here is fine, too. 👍